core/block_svg.js

```diff
@@ -687,52 +688,18 @@ Blockly.BlockSvg.prototype.setCollapsed = function(collapsed) {
 
 /**
  * Open the next (or previous) FieldTextInput.
- * @param {Blockly.Field|Blockly.Block} start Current location.
+ * @param {!Blockly.Field} start Current field.
  * @param {boolean} forward If true go forward, otherwise backward.
  */
 Blockly.BlockSvg.prototype.tab = function(start, forward) {
-  var list = this.createTabList_();
-  var i = start != null ? list.indexOf(start) : -1;
-  if (i == -1) {
-    // No start location, start at the beginning or end.
-    i = forward ? -1 : list.length;
-  }
-  var target = list[forward ? i + 1 : i - 1];
-  if (!target) {
-    // Ran off of list.
-    var parent = this.getParent();
-    if (parent) {
-      parent.tab(this, forward);
-    }
-  } else if (target instanceof Blockly.Field) {
-    target.showEditor();
-  } else {
-    target.tab(null, forward);
-  }
-};
+  var tabCursor = new Blockly.TabNavigateCursor();
```

Comment:
At first glance I don't see a reason to keep it around. All the cursor is really doing is holding the curNode which it looks like we are going to reset anyway. And we don't do anything expensive in the constructor so doesn't seem like it would be expensive to recreate every time.

Reply:
Sounds good.